### ISSUE TITLE
text for progess

### ISSUE THREAD

**Issue by niccokunzmann** [CoderDojoPotsdam/regex-tutorial#11]
When the mouse hovers above the bubbles on the progress bar, the title of the linked page should be displayed.

**Comment by weerox** [comment]
Do you want the built-in hover title design, or should we create something of our own?
I like the GitHub implementation (to view this, hover the "Owner" tag in a comment).

**Comment by niccokunzmann** [comment]
Looks good. As you like. 

I think, the more we do although it is fun, the less new people can do and maybe the less contributions we get. How did you decide to contribute? Was it because you fixed a small mistake? If we also find those small issues, we can get more people involved.

**Comment by weerox** [comment]
Yeah, you're right.

**Comment by joshhazelhurst123** [comment]
Hi, I am interested in contributing to this project and learning as much as possible. I like the github implementation as mentioned by @weerox. Is there a github api to search through or code for the tag?

**Comment by niccokunzmann** [comment]
In Firefox (Chrome, Opera and Edge, too) You can inspect websites and tags. You can right-click on the page, click on Inspect Element and then have a look at source code and styling.
![grafik](https://user-images.githubusercontent.com/564768/28617148-884653e2-71ff-11e7-9b3f-20479120b1e5.png)

It seems like it uses CSS animations to fade in after 0.4 seconds:
![grafik](https://user-images.githubusercontent.com/564768/28617104-55cef8c4-71ff-11e7-8c29-be0e284ba4d9.png)

**Comment by niccokunzmann** [comment]
And you can view the animation and the styling by clicking on the file.
![grafik](https://user-images.githubusercontent.com/564768/28617241-ecba2222-71ff-11e7-87a1-542005068201.png)

I do not know if I could answer your question in any way.

**Comment by joshhazelhurst123** [comment]
hmmm I can't get edit the progress bar, I look at progress.css and tutorial.html to make the changes that I wanted to make on the progress bar but I can't figure out how it works. Thanks.

**Comment by niccokunzmann** [comment]
@joshhazelhurst123 Hi, did you [set up jekyll](https://github.com/CoderDojoPotsdam/regex-tutorial#install-jekyll) locally?

The code for the progress bar starts here:
https://github.com/CoderDojoPotsdam/regex-tutorial/blob/058c263dc6d11b9d45ed63096f9c5006456d2d7f/_layouts/tutorial.html#L70
and this is the link for which the hover effect can be changed:
https://github.com/CoderDojoPotsdam/regex-tutorial/blob/058c263dc6d11b9d45ed63096f9c5006456d2d7f/_layouts/tutorial.html#L96

Could I answer some questions? Do you have new questions? Which changes did you try and make?

**Comment by joshhazelhurst123** [comment]
Yes jekyll works, you have helped me a lot! 

Thanks for your comment and reply. I managed to put a basic hover tag in tutorial.html which hovers over all paragraphs, and a href links.  
  p:hover, a:hover {
  background-color: yellow;
} 
I would like to make the changes as requested in this issue request, I want the tooltip to appear when hovering over the link in the progress bar.
I think adding the tooltip-appear in tutorial.html under <style> tab might work. Or in progress.css but the next part I am not sure about.
How can I make the tooltip appear when hovering over the link in the progress bar and not over all the links? 

Thanks again.

![untitled](https://user-images.githubusercontent.com/30138699/53221207-46785e80-36cd-11e9-8b8f-34b036db2449.jpg)

**Comment by niccokunzmann** [comment]
I think, you may be able to use the id or the class selector, not only the tag selector like "p" or "a".
Maybe [this](https://www.w3schools.com/Css/css_syntax.asp) clarifies the difference.

https://github.com/CoderDojoPotsdam/regex-tutorial/blob/058c263dc6d11b9d45ed63096f9c5006456d2d7f/_layouts/tutorial.html#L96
generates 
```html
<a id="page-de/03-03.html" class="step" href="../de/03-03.html"></a>
```
so you can use class and step.

**Comment by joshhazelhurst123** [comment]
Thanks, so I have working,
.step:hover{
  background-color: yellow;
} 
The next part is adding the tooltip on :hover 
I am thinking, 
.step:tooltip-appear{
 aria-label="This user is a member of the CoderDojo Potsdam organization."
} 
<></> closing on brackets below, (can't add them in, they get picked up as links)
 div class="links"
 a href="https://github.githubassets.com/app/node_modules/@primer/css/tooltips/tooltips.scss"a
How does that look to you? What am I missing? Thanks

**Comment by joshhazelhurst123** [comment]
https://www.w3schools.com/howto/tryit.asp?filename=tryhow_css_tooltip
https://www.w3schools.com/css/css_tooltip.asp
are my resources I'm looking at. The span, div tag are confusing me.

**Comment by niccokunzmann** [comment]
You can make code comments by selecting the text and using the <> button above.
```html
<a
    href="http://test.com">test</a>
```

**Comment by niccokunzmann** [comment]
> The span, div tag are confusing me.

Hm. I do not know how that expresses itself. I think, for confusion, one has an inconsistent model with at least two different conflicting strains of thought. Could you write down all of them, so I can see what is there? Currently, I could not phrase it better into the uncertain than the tutorial.

**Comment by joshhazelhurst123** [comment]
Ah Yes you have read my mind. There are two different conflicting strains of thought. I am using w3school online tutorials and I am looking at your example about implementing the tooltip on the regextutorial site. On the w3schools example of tooltip, they use a span and div tag to implement the tooltip, with CSS defining the tooltip. 
Then trying to apply what I have learnt from w3schools here, I don't want to add a div and span tag into tutorial.html but would rather use the id or the class selector, not only the tag selector like "p" or "a". As you mentioned earlier. 

In summary, can I implement the tooltip from w3schools without the div and span tags. Hopefully you understand me. Thank you.

**Comment by niccokunzmann** [comment]
> can I implement the tooltip from w3schools without the div and span tags

I think, you can do something like this:
```html
<span id="page-de/03-03.html" class="step">
    <a href="../de/03-03.html"></a>                    
    <span class="tooltip">Tooltip text</span>
</span>
```

span and div are only slightly different. Using CSS, you can convert one into the other. a span differs form div in values of some attributes.
For CSS, span or div or jahdkjahsdkhasd can be used to style `<span>`, `<div>` and `<jahdkjahsdkhasd>` elements from html. Hm, so I think, you will need to edit some of the code and add new tags for the new structure but you can choose whether you like to have span or div or anything else.

**Comment by joshhazelhurst123** [comment]
![image](https://user-images.githubusercontent.com/30138699/53468629-e2380f00-3abf-11e9-8a82-1bce1f1c5e6a.png)

I have a tooltip connected as shown above. Not sure why the progress bar has halfway disappeared.  I updated tutorial.html with the following code. 

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/css/bootstrap.min.css">
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/js/bootstrap.min.js"></script>

    <script>
$(document).ready(function(){
  $('[data-toggle="tooltip"]').tooltip({title: "Hooray", placement: "bottom"});
});
</script>

And I put in the a tags
data-toggle="tooltip" title="Hooray!"

**Comment by niccokunzmann** [comment]
Can you, instead of sending the code in the issue, create a Pull Request
with the code?
This way, I can exactly see what is going on and we can discuss it line
by line. This is the workflow we usually take.

> <script
src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
<script
src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/js/bootstrap.min.js"></script>

I will not add these libraries and we can work later around them.
It is not necessary to include megabytes of code and break offline
usability.

**Comment by joshhazelhurst123** [comment]
pull request. #119 
Creating a tooltip then merging it with the progress bar was unsuccessful for me. I then added in the bootstrap libraries for a tooltip in order to produce the tooltip as shown above. I understand that you wouldn't want them as they would include megabytes to the website.
